src/sql/workbench/contrib/commandLine/electron-browser/commandLine.ts

```diff
@@ -324,9 +368,26 @@ export class CommandLineWorkbenchContribution implements IWorkbenchContribution,
 		profile.setOptionValue('applicationName', applicationName);
 		profile.setOptionValue('databaseDisplayName', profile.databaseName);
 		profile.setOptionValue('groupId', profile.groupId);
+		// Set all advanced options
+		let advancedOptions = this.getAdvancedOptions(args.options, profile.getOptionKeyIdNames());
+		advancedOptions.forEach((v, k) => {
+			profile.setOptionValue(k, v);
+		});
 		return this._connectionManagementService ? this.tryMatchSavedProfile(profile) : profile;
 	}
 
+	private getAdvancedOptions(options: string, idNames: string[]): Map<string, string> {
+		let advancedOptionsMap = new Map<string, string>();
+		if (options) {
+			JSON.parse(options, (k, v) => {
```

Comment:
Would suggest surrounding this with a try catch and displaying a clear error to the user if parsing fails (and then probably exiting the flow so we don't try to connect with unexpected options)